.github/workflows/ci.yml

```diff
@@ -149,3 +149,21 @@ jobs:
           toolchain: stable
       - uses: Swatinem/rust-cache@v2
       - run: cargo lint
+
+  build-docs:
+    name: Test Building Docs
+    runs-on: ubuntu-latest
+    env:
+      MDBOOK_VERSION: 0.4.31
+    steps:
+      - uses: dtolnay/rust-toolchain@439cf607258077187679211f12aa6f19af4a0af7
+        with:
+          toolchain: stable
+      - uses: Swatinem/rust-cache@v2
```

Comment:
this is a nitpick, but I think we should stick to hashes when it goes for non-official actions like this one; tag can be easily overwritten by a maintainer and injected with mallicious code. I understand that someone already did the same in this file (I mean marked this action with a tag), so I won't block the review, but this is something that in my opinion should be changed (again - not necessarily in this PR).

Reply:
Noticed it as well, but thought maybe there was already some discussion about it previously, since it's used in a few places in the file - will update to point to a commit 👍